auth/password_hash_test.go

```diff
@@ -64,3 +65,64 @@ func TestSetBcryptCost(t *testing.T) {
 	assert.Equal(t, bcryptDefaultCost, bcryptCost)
 	assert.True(t, bcryptCostChanged)
 }
+
+func BenchmarkCompareHashAndPassword(b *testing.B) {
+	// Get the bcrypt hash of the given password.
+	bcryptHash := func(password []byte) []byte {
+		hash, err := bcrypt.GenerateFromPassword(password, bcryptDefaultCost)
+		require.NoError(b, err)
+		return hash
+	}
+	// hashAndPassword represents the password hash pair.
+	type hashAndPassword struct {
+		hash     []byte
+		password []byte
+	}
+	// Setup cache of size N
+	const numPasswords = 100
+	cachedHashes = NewCache(numPasswords)
```

Comment:
Overwriting the global cache doesn't seem safe.

Is it feasible to move `compareHashAndPassword` to be a method on a cache?
There's only 1 use in the code, which can still just use the global variable, but allows tests to use their own cache.

Reply:
Yeah, it is feasible to attach compareHashAndPassword to the cache and the change is included in the lates commit.